internal/mutators/namespace.go

```diff
@@ -52,19 +53,66 @@ func (m *Namespace) Handle(ctx context.Context, req admission.Request) admission
 // Currently, we only add `included-namespace` label to non-excluded namespaces
 // if the label is missing.
 func (m *Namespace) handle(log logr.Logger, ns *corev1.Namespace) {
-	// Early exit if the namespace is excluded.
-	if config.ExcludedNamespaces[ns.Name] {
-		return
-	}
 
-	// Add label if the namespace doesn't have it.
-	if _, hasLabel := ns.Labels[api.LabelIncludedNamespace]; !hasLabel {
-		if ns.Labels == nil {
-			ns.Labels = map[string]string{}
+	// if includedNamespaces is set, we dont need excluded as everyone that is not included is
+	// automatically excluded.
+	// if includedNamespaces is set, we need to set the label only to the included namespaces list
+
+	// if parameter included-namespaces is set, everything else is excluded
+	// so, we set the annotation only if ns exists in the map
+
+	//log.Info("we are in the mutator handle and this is the excluded map:", config.ExcludedNamespaces)
+
+	if config.IncludedNamespaces != nil {
+
+		// work with regulars
+		//log.Info("we are in the mutator handle, inside included process", config.IncludedNamespaces)
+
+		for nsRegexp, _ := range config.IncludedNamespaces {
+
+			//log.Info("this is the ns in the included map", nsRegexp)
+			if _, ok := config.ExcludedNamespaces[ns.Name]; ok {
```

Comment:
Why is this in the loop? This does not seem to use regex, could be outside it.
When outside, we can notice this is in both if / else condition, so it can be moved outside of both and unified.